### ISSUE TITLE
Using a dedicated authenticated proxy for a Winforms .net 4.8 application

### ISSUE THREAD

**Issue by JimWard2018** [MicrosoftEdge/WebView2Feedback#2532]
If I don't use a WebProxy my application is working.
My development environment is currently Windows 10 Enterprise version 21H2
My CoreWebView2InitializationCompleted has a BrowserVersionString: 104.0.1288.0 canary

My application uses the WebProxy class
`https://docs.microsoft.com/en-us/dotnet/api/system.net.webproxy?view=netframework-4.8`
to set up it's own authenticated proxy independent of any windows default web proxy that may be defined.

I'm using the approach suggested here to get WebView2 working with a proxy

`https://stackoverflow.com/questions/66254963/autofill-proxy-requirements-webview2`

1) Specifying --proxy-server in AdditionalBrowserArguments
2) Using a CoreWebView2  BasicAuthenticationRequested event to supply the required credentials.

That works in that I get passed the logon form and I can see the authentication event was triggered and the credentials were supplied successfully.

After that my application loads a home and project dashboard view; each of which uses a WebView2 control.
Each of these views is returning html to displaying items to click on/open and work on 

Before using the proxy I had problems with these two initially as 'sometimes' one or other wouldn't display anything and I could see no obvious errors.
The events for the WebView2 controls were intertwined and I wondered whether that was the problem.
So I changed the code so it loaded one, before loading the other and that seemed to fix the problem

Now using The WebProxy I have a hard issue of one the dashboards not loading

I've attached a log of the current sequence of events

Both dashboards navigate to the same jsp i.e.  

`https://site601-instance601.smartcommunications.cloud/one/client-dashboard/web/dashboardsc2.jsp`

but with a different parameter; so I expected and see only one authentication 

The waiting to load messages is my attempt at loading them in parallel (as I explained earlier)

I would be very grateful if you could help the following concerns that I have

1) Have you had issues when WebView2 controls are loading at the same time; if so do you have specific advice on how to deal with that scenario.
2) How do I deal with the WebErrorStatus=ValidProxyAuthenticationRequired at 16:37:23,224 in the debug trace when the authentication should have worked for that (it worked at the logon page in another WebView2 control
3) Do you have any advice on using the system WebProxy class and passing --proxy-server in AdditionalBrowserArguments

**Comment by JimWard2018** [comment]
[](url
[debug.txt](https://github.com/MicrosoftEdge/WebView2Feedback/files/8920592/debug.txt)
)

**Comment by JimWard2018** [comment]
I don't know how I managed to close this but that was a mistake!

**Comment by JimWard2018** [comment]
I noticed that version 1.0.1245.22 of WebView2 was available so I've updated to that but it made no difference.
I've also changed my code to throw away the Webview2 control that fails with ValidProxyAuthenticationRequired and use a new one but that doesn't help either.
I've attached the debug of this run too

**Comment by JimWard2018** [comment]
[debug2.txt](https://github.com/MicrosoftEdge/WebView2Feedback/files/8928936/debug2.txt)

**Comment by JimWard2018** [comment]
Using this search https://github.com/MicrosoftEdge/WebView2Feedback/search?p=2&q=WebView2+blank+not+working&type=issues
I can see quite a few others have problems with WebView2 not displaying properly sometimes and that it's a known problem with no fix yet.

After reading one of those I added a **CoreWebView2 ProcessFailed event handler** but unfortunately that was never called in my case

I also saw that using additional browser arguments **--ignore-gpu-blocklist** **--disable-gpu-driver-bug-workarounds** can be useful
So I added those additional browser arguments and my application started working

**This is a hard fault for me which happens every time (without those additional browser arguments)**

**Comment by JimWard2018** [comment]
From https://github.com/MicrosoftEdge/WebView2Feedback/issues/2276 that indicates that the problem is a graphics driver issue.

I've attached the DxDiag output of the Windows 2016 Server VM that has this problem

**Comment by JimWard2018** [comment]
@yildirimcagri do you have any advice for me? Do you need any more information? I'm running out of things to try.

**Comment by JimWard2018** [comment]
[WinInetInterop.txt](https://github.com/MicrosoftEdge/WebView2Feedback/files/8950049/WinInetInterop.txt)

I had to change the extension from vb to txt for it to load here

**Comment by JimWard2018** [comment]
@champnic Do you have any suggestions as to what my next move should be?

**Comment by JimWard2018** [comment]
It generates this crash pad report
[3ed55b8b-5231-4858-8c15-c01c84efe208.zip](https://github.com/MicrosoftEdge/WebView2Feedback/files/8950554/3ed55b8b-5231-4858-8c15-c01c84efe208.zip)

**Comment by yildirimcagri** [comment]
Hi @JimWard2018 thanks for all the information. This could be a bug in WebView2 with storing proxy credentials across two different WebViews. It's possible that there is no visible UI in the first WebView2 because there is no successful navigation, but it is strange that the first WebView would fail and the second succeeds when the proxy credentials would be prompted and supplied in the first WebView. The dump in crash pad report seems unrelated.

I do have a couple questions for you:
1) Could you provide the log differentiating the events between the two WebViews? As it is, it's hard to tell which event corresponds to which WebView.
2) Is the "ProxyAuthorization" log the place you get the BasicAuthenticationRequested event? If it is, looking at the logs, it seems sometimes the BasicAuthenticationRequested event comes after the NavigationCompleted with ValidProxyAuthenticationRequired. For the proxy authentication to be prompted, the first navigation request will fail with the 407 response. Could the first NavigationCompleted event with ValidProxyAuthenticationRequired be for that failed navigation? 
3) OperationCanceled would generally be reported if the WebView2 navigates away before completing load (such as by running script during load). Is that what's happening here? I do see a successful NavigationCompleted event in the end that could be the result of the final navigation for the second WebView.
4) What happens if you try the same scenario in two different tabs in the browser, do you get two proxy authentication prompts or does the second tab navigate using stored credentials?

**Comment by JimWard2018** [comment]
To help your understanding here's some screen shots

My application starts up with a logon form containing the first WebView2 control

![1 WebView2 Logon](https://user-images.githubusercontent.com/40032214/175021466-623d24a6-3713-4795-b676-2b0f73f44690.png)

It main form then loads initially trying to display the Home Dashboard which contains the second WebView2 control
The application waits for a the Home Dashboard WebView2 to get a successful NavigationCompleted event but gives up after 20 seconds

![2 WebView2 Home Dashboard](https://user-images.githubusercontent.com/40032214/175021831-4cf59b49-b02c-40c9-a4cd-75a7ed400ffc.png)

The application then tries to display the Project Dashboard with a third WebView2 using the same CoreWebView2Environment and the same common code and is successful

![3 WebView2 Project Dashboard](https://user-images.githubusercontent.com/40032214/175022656-56829477-2e2d-4b7f-a742-60f6a6aa4867.png)

**Comment by JimWard2018** [comment]
The following debug file shows the above process going on and I've added extra comments in it to make clear which of the above forms/tabs the debug information applies to and what code it is executing

[debug3.txt](https://github.com/MicrosoftEdge/WebView2Feedback/files/8957276/debug3.txt)

**Comment by JimWard2018** [comment]
It's difficult to replicate this behaviour in a browser but I would expect the first time that the proxy is used that authentication is required; with stored credentials being used in subsequent browser navigation using the proxy

**Comment by JimWard2018** [comment]
I have one remaining question from my original questions and that is ...

Is there an additional browser argument that maps onto BypassOnLocal which Windows and the WebProxy class have as parameters?

**Comment by yildirimcagri** [comment]
Glad you were able to figure out the issue. For being able to bypass local resources in proxy, I'm not sure if there is an equivalent flag, but could you specify the [--proxy-bypass-list ](https://www.chromium.org/developers/design-documents/network-settings/) with the schemes/urls that you don't want WebView2 to use the proxy for?

**Comment by JimWard2018** [comment]
@yildirimcagri Yes I'm already using --proxy-bypass-list and that works. It's just that windows proxy settings from within the Chrome browser and in the .Net 4.8 WebProxy class have the by pass local flag and so my Winforms application allowed it and the by pass list to be set. 

Opening Edge and going to its windows proxy settings I see that flag is there but disabled. I guess that means that it's not supported

![EdgeProxySettings](https://user-images.githubusercontent.com/40032214/176389648-e40ab41d-6534-4264-b341-c8cf3eff5da8.JPG)
